### ISSUE TITLE
onebusaway-transit-data-federation-builder fails if agency ID contains underscore

### ISSUE THREAD

**Issue by rkerr** [OneBusAway/onebusaway-application-modules#140]
Precache will fail to build with org.onebusaway.exceptions.NoSuchStopServiceException

Seems to have been discussed before:

https://groups.google.com/d/msg/onebusaway-developers/RHHfZy_uLJM/mTklBXweYKUJ

This is rather unfortunate when trying to use the UK traveline data as all their operator IDs are preffixed with 'OId_'

**Comment by sheldonabrown** [comment]
Yeah, that is a shame.  However, the agency + "_" is pretty well baked into
OneBusAway.  Any chance you could run the gtfs transformer over the GTFS
first to clean up the operator ids before building?

http://developer.onebusaway.org/modules/onebusaway-gtfs-modules/current-SNAPSHOT/onebusaway-gtfs-transformer-cli.html

Sheldon

**Comment by rkerr** [comment]
I certainly can process the GTFS before processing. If it's not going to be possible to change this I think catching it earlier and providing a more obvious error message may be helpful. Getting a NoSuchStopServiceException in the precache phase is rather confusing and it took quite a while to get to the bottom of what was going on.

**Comment by ethanpooley** [comment]
We just ran into this when trying to build a bundle for Delaware DART. The `agency_id` in that feed is `DART_DE`. We're handling it manually, but it confused us for a bit. Just mentioning it here to upvote the issue.

Delaware DART feed: http://transitfeeds.com/p/dart-first-state/209/latest
